genson/generator.py

```diff
@@ -19,7 +19,7 @@ class Schema(object):
     with existing schemas and objects before being serialized.
     """
 
-    def __init__(self, merge_arrays=True):
+    def __init__(self, merge_arrays=True, _c=None):
```

Comment:
Instead of passing a private context argument here, I would just convert everything to `**options`, lazily set some defaults, and then pass it down the chain. That way there's no protected interface that users have to figure out and work around.